apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/RowFilters.cs

```diff
@@ -0,0 +1,407 @@
+            throw new InvalidOperationException();
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="RowFilter"/> instance which matches only the first
+        /// N cells of each row.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// If duplicate cells are present, as is possible when using an Interleave,
+        /// each copy of the cell is counted separately.
+        /// </para>
+        /// </remarks>
+        /// <param name="limit">
+        /// The maximum number of cells allowed per row.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter CellsPerRowLimit(int limit)
+        {
+            throw new InvalidOperationException();
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="RowFilter"/> instance which skips the first N cells
+        /// of each row, matching all subsequent cells.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// If duplicate cells are present, as is possible when using an Interleave,
+        /// each copy of the cell is counted separately.
+        /// </para>
+        /// </remarks>
+        /// <param name="limit">
+        /// The maximum number of cells allowed per row.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter CellsPerRowOffset(int limit)
+        {
+            throw new InvalidOperationException();
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="RowFilter"/> instance which sends rows through
+        /// several RowFilters in sequence.
+        /// </summary>
+        /// <param name="filters">
+        /// The elements of "filters" are chained together to process the input row:
+        /// in row -> f(0) -> intermediate row -> f(1) -> ... -> f(N) -> out row
+        /// The full chain is executed atomically.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter Chain(params RowFilter[] filters)
+        {
+            throw new InvalidOperationException();
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="RowFilter"/> instance which matches only cells from
+        /// columns whose qualifiers satisfy the given RE2 regex.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Note that, since column qualifiers can contain arbitrary bytes, the `\C`
+        /// escape sequence must be used if a true wildcard is desired. The `.`
+        /// character will not match the new line character `\n`, which may be
+        /// present in a binary qualifier.
+        /// </para>
+        /// </remarks>
+        /// <param name="regex">
+        /// The RE2 regex used to match column qualifiers.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter ColumnQualifierRegex(ColumnQualifier regex)
```

Comment:
Overload to accept a string for the regex?

Reply:
+1

Reply:
`string` is implicitly convertible to `ColumnQualifier`. I specifically added `RowKey` and `ColumnQualifier` to reduce the number of overloads required, especially in cases where multiple keys, qualifiers were needed. The only case where I haven't done this is for the cell values, and it results in 8 overloads of `Cell.ValueRange` instead of 4 (it really should be 16 to cover all cases, but I think what I have should be sufficient).

Reply:
Ah, that makes sense. It's probably worth mentioning that in remarks just for clarity - eventually. (I'm very aware this is just an early stage skeleton.)

Reply:
Done (and also anywhere else `ColumnQualifier` is used).

Reply:
...and `RowKey` as well.